Add support for track descriptions

cdamus opened the pull request:
Add a description to the track data. Try to load it, where the schema has one, from the database. Otherwise provide some interpretive text. The latter descriptions are derived in a few different ways:

- database schema descriptions in the Perfetto documentation
- documentation from related systems (e.g., proc(5) manpage)
- mimic AGI (e.g., the display of the number of threads in a process)

Also fix the build script to account for the output directory that it needs to clean now being a real directory and not a symbolic link.

cdamus commented on ui/tsconfig.json:
```diff
@@ -18,5 +18,6 @@
     },
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
+    "skipLibCheck": true,
```
When I try to build Perfetto UI on its own, not in context of the Theia application that submodules it, I get reams of spurious compilation errors in the `node_modules/` dependencies. This stops that, but perhaps I should not commit this change for everyone else if they haven't got this problem. I'll revert it.

jfaltermeier replied:
I think we can leave it in, I was just curious.

cdamus commented on ui/src/controller/track_decider.ts:
```diff
@@ -202,6 +202,93 @@ class TrackDecider {
+      }>): string | undefined {
+    const {upid, processName, numThreads} = args;
+
+    const hasProcessName = !!processName && upid != undefined;
```
On another project (which has the lint rule configured that you mention) I was advised in the same situation to use this formulation because it also covers the case of `null`. i.e., `upid != undefined` is equivalent to `upid !== undefined && upid !== null`. It seems that ESLint is aware of this special case of the other otherwise deprecated `!=` operator and so doesn't flag it.

However, I am wary of the seeming inconsistency/magic involved and will prefer the more explicit, if more verbose, formulation. So I'll change it. Thanks!

cdamus commented:
Thanks for the review, @jfaltermeier ! Commit d3919804f4ca1203fac1acc480e153cbefb37126 addresses the comments.

I did choose to revert the tsconfig change because I have no evidence that it solves a problem for anyone but me.
